pkg/sidecar_mounter/sidecar_mounter.go

```diff
@@ -349,6 +368,17 @@ func getAudienceFromContextAndIdentityProvider(ctx context.Context, identityProv
 }
 
 func StartTokenServer(ctx context.Context, tokenURLSocketPath string, identityProvider string) {
+	// Clean up any stale socket file before creating a new one.
+	if _, err := os.Stat(tokenURLSocketPath); err == nil {
```

Comment:
please create a helper function to check a socket path and delete the socket. this same block is used to delete metrics socket